carma_wm/src/CARMAWorldModel.cpp

```diff
@@ -0,0 +1,333 @@
+
+    std::vector<std::tuple<size_t, std::vector<double>>> vec;
+
+    lanelet::BasicPoint2d prevEndPoint;
+    lanelet::BasicPoint2d first_2d, second_2d, third_2d;
+    
+    for (size_t n = 0; n < lanelets.size(); n++) {
+      lanelet::ConstLanelet ll = lanelets[n];
+      auto mutableCenterLine = lanelet::utils::to2D(ll.centerline()).basicLineString();
+
+      if (mutableCenterLine.empty()) {
+        throw std::invalid_argument("Provided lanelet contains no centerline");
+      }
+
+      for (size_t i = 0; i < mutableCenterLine.size() - 1; i++) { 
+        
+        // If the first lanelet initialize the 3 points needed for computing curvature
+        if (n == 0 && i <= 1) { 
+          if (i == 0) {
+            std::vector<double> curvatures;
+            vec.push_back(std::make_tuple(n, curvatures));
+            continue;
+          } else { // i == 1
+            first_2d = mutableCenterLine[i - 1];
+            second_2d = mutableCenterLine[i];
+            third_2d = mutableCenterLine[i + 1]; 
+            std::get<1>(vec.back()).push_back(computeCurvature(first_2d, second_2d, third_2d));
+            continue;
+          }
+        } else if (n != 0 && i <= 1) {
+            bool new_segment_needed = lanelet::geometry::distance2d(prevEndPoint, mutableCenterLine[i]) > 0.1;
+            if (new_segment_needed) {
+              if (i == 0) {
+                // Consider as disjoint lanelet
+                std::vector<double> curvatures;
+                vec.push_back(std::make_tuple(n, curvatures));
+                continue;
+              } else { // i == 1
+                first_2d = mutableCenterLine[i - 1];
+                second_2d = mutableCenterLine[i];
+                third_2d = mutableCenterLine[i + 1];
+                std::get<1>(vec.back()).push_back(computeCurvature(first_2d, second_2d, third_2d));
+                continue;
+              }
+            }
+        }
+
+        first_2d = second_2d;
+        second_2d = third_2d;
+        third_2d = mutableCenterLine[i + 1];
+        std::get<1>(vec.back()).push_back(computeCurvature(first_2d, second_2d, third_2d));
+      }
+
+      prevEndPoint = mutableCenterLine.back();
+
+    }
+
+    return vec;
+  }
+  
+  lanelet::LaneletMapConstPtr CARMAWorldModel::getMap() const {
+    return std::static_pointer_cast< lanelet::LaneletMap const >(semantic_map_); // Cast pointer to const variant
+  }
+
+  LaneletRouteConstPtr CARMAWorldModel::getRoute() const {
+    return std::static_pointer_cast< const lanelet::routing::Route >(route_); // Cast pointer to const variant
+  }
+
+  void CARMAWorldModel::setMap(lanelet::LaneletMapPtr map) {
+    semantic_map_ = map;
+    // Build routing graph from map
+    lanelet::traffic_rules::TrafficRulesUPtr traffic_rules = lanelet::traffic_rules::TrafficRulesFactory::create(lanelet::Locations::Germany, lanelet::Participants::VehicleCar);
```

Comment:
It is the only one that currently exists, and I think it will work for our current use case. I will look into making a user story to flesh out a US one.